brouter-core/src/main/java/btools/router/RoutingEngine.java

```diff
@@ -669,13 +670,13 @@ int getRandomDirectionByRouting(OsmNodeNamed wp, double searchRadius) {
 
     switch (preferredRandomType) {
       case AreaInfo.RESULT_TYPE_ELEV50:
-        ais.sort(Comparator.comparingInt(AreaInfo::getElev50Weight));
+        Collections.sort(ais, Comparator.comparingInt(AreaInfo::getElev50Weight));
```

Comment:
@afischerdev `Comparator.comparingInt` also requires Android 7.

Reply:
@afischerdev BRouter app has `minSdkVersion 21` so IDEA should inform for all these?

Reply:
Using CLI with gradlew for building my test lib.
In the Android Studio same source, `comparingInt` is marked (api 24). When changed the Comparator, the `Collections.sort` is marked and recommended to use List.sort(). After doing this, it's marked as wrong api 21.
Intelligent development systems are sometimes funny.